From bffdc2f1930d82478d41d666945c93d78b43a963 Mon Sep 17 00:00:00 2001
From: Jared McArthur <j-mcarthur@ti.com>
Date: Mon, 29 Jul 2024 13:47:55 -0500
Subject: [PATCH] arm64: dts: ti: k3-j722s: Add gpio-reserved-ranges for
 main_gpio1

Commit ed07d82f9e3e8 ("arm64: dts: ti: k3-am62p-j722s: Move SoC-specific
node properties") introduced the main_gpio1 node and included the
ngpios, but missed the gpio-reserved-ranges property. As a result, the
driver could try to access gpios that do not exist. Fix this by
introducing the gpio-reserved-ranges property.

Fixes: ed07d82f9e3e ("arm64: dts: ti: k3-am62p-j722s: Move SoC-specific node properties")
Signed-off-by: Jared McArthur <j-mcarthur@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j722s-main.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/ti/k3-j722s-main.dtsi b/arch/arm64/boot/dts/ti/k3-j722s-main.dtsi
index c797980528ec..8f279261b742 100644
--- a/arch/arm64/boot/dts/ti/k3-j722s-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j722s-main.dtsi
@@ -214,4 +214,5 @@ &main_gpio1 {
 	gpio-ranges = <&main_pmx0 7 101 25>, <&main_pmx0 42 137 5>,
 			<&main_pmx0 47 143 3>, <&main_pmx0 50 149 2>;
 	ti,ngpio = <73>;
+	gpio-reserved-ranges = <0 7>, <32 10>;
 };
-- 
2.34.1

